src/components/drag-and-drop-container.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+
+function DragAndDropContainer({
+  items,
+  droppableId,
+  onItemMove
+}) {
+
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragStart = () => {
+    setIsDragging(true);
+  };
+
+  const handleDragEnd = ({ source, destination }) => {
+    setIsDragging(false);
+    const fromIndex = source?.index;
+    const toIndex = destination?.index;
+    if (!!source && !!destination && fromIndex !== toIndex && toIndex >= 0 && toIndex <= (items.length - 1)) {
+      onItemMove(fromIndex, toIndex);
+    }
+  };
+
+  return (
+    <DragDropContext onDragStart={handleDragStart} onDragEnd={handleDragEnd}>
+      <Droppable droppableId={droppableId} ignoreContainerClipping>
+        {droppableProvided => (
+          <div ref={droppableProvided.innerRef} {...droppableProvided.droppableProps}>
+            {items.map((item, index) => (
+              <Draggable key={item.key} draggableId={item.key} index={index}>
+                {(draggableProvided, draggableState) => (
+                  <div
+                    key={item.key}
+                    ref={draggableProvided.innerRef}
+                    {...draggableProvided.draggableProps}
+                    style={{
+                      userSelect: draggableState.isDragging ? 'none' : null,
+                      ...draggableProvided.draggableProps.style
+                    }}
+                    >
+                    <section key={item.key}>
```

Comment:
Just out of curiosity, is there a reason to nest another `section` inside the `div`, it seems to not have any purpose?

Reply:
I've done that so that the `key` is set in a clear way on the iterated elements. Otherwise that needed to go inside the rendered component and it's easier to miss to set it. Found that it gets a problem out of the way and contained in just in one place.

Reply:
Just nitpicking here, but I think we could reduce the nesting here as the key is set already on the parent `div`.